ProcessMaker/Jobs/BpmnAction.php

```diff
@@ -154,45 +152,44 @@ public function withUpdatedContext(callable $callable)
      *
      * @return ProcessRequest
      */
-    protected function lockInstance($instanceId)
+    private function lockInstance($instanceId)
     {
         try {
             $instance = ProcessRequest::findOrFail($instanceId);
             if (config('queue.default') === 'sync') {
                 return $instance;
             }
             if ($instance->collaboration) {
-                $ids = $instance->collaboration->requests->pluck('id')->toArray();
+                $ids = $instance->collaboration->requests()->where('status', 'ACTIVE')->pluck('id')->toArray();
             } else {
                 $ids = [$instance->id];
             }
             $lock = $this->requestLock($ids);
             // If the processes are going to have thousands of parallel instances,
             // the lock will be released after a while.
-            $timeout = config('app.bpmn_actions_max_lock_timeout', 6000) ?: 6000;
+            $timeout = config('app.bpmn_actions_max_lock_timeout', 60000) ?: 60000;
             $interval = config('app.bpmn_actions_lock_check_interval', 1000) ?: 1000;
-            $maxRetries = ceil($timeout / $interval * 1000);
+            $maxRetries = ceil($timeout / $interval);
             for ($tries=0; $tries < $maxRetries; $tries++) {
                 $currentLock = $this->currentLock($ids);
                 if (!$currentLock) {
                     if (ProcessRequest::find($instanceId)) {
                         $lock = $this->requestLock($ids);
                     } else {
-                        return false;
+                        throw new Exception('Unable to lock instance #' . $this->instanceId . ': Request does not exists');
                     }
                 } elseif ($lock->id == $currentLock->id) {
                     $instance = ProcessRequest::findOrFail($instanceId);
                     $this->activateLock($lock);
                     return $instance;
                 }
                 // average of lock time is 1 second
-                usleep($interval);
+                usleep($interval * 1000);
```

Comment:
As normally it will take more than 1 second, it can be better to use just  "sleep" function. Check:
https://www.php.net/manual/en/function.usleep.php
The section with the note: _Values larger than 1000000 (i.e. sleeping for more than a second) may not be supported by the operating system._

Reply:
Good catch, but `app.bpmn_actions_lock_check_interval` was defined in milliseconds

Reply:
Fixed